### DISCUSSION TITLE
pip install polars-xdt fails - macOS Sonoma 14.3.1

### DISCUSSION THREAD

**Discussion by Chretien** [MarcoGorelli/polars-business#59] [General]
Hello,

I tried to install polars-xdt through pip and was met with the following error code and am not able to successfully install this package. Please let me know what other information I can give.

I am working on a MacBook Pro M1 Pro running Sonoma 14.3.1. My pip version is 24.0, and I'm running this on Python 3.9.10.

❯ pip install polars-xdt
WARNING: Ignoring invalid distribution -astapi (.../python3.9/site-packages)
Collecting polars-xdt
  Downloading polars_xdt-0.12.2.tar.gz (976 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 976.1/976.1 kB 14.7 MB/s eta 0:00:00
  Installing build dependencies ... error
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [5 lines of output]
      Collecting maturin<2.0,>=1.0
        Downloading maturin-1.4.0-py3-none-macosx_10_12_x86_64.macosx_11_0_arm64.macosx_10_12_universal2.whl.metadata (18 kB)
      ERROR: Ignored the following yanked versions: 0.13.32, 0.13.33, 0.13.35, 0.13.41, 0.13.42, 0.13.43, 0.13.45, 0.13.47, 0.13.48, 0.13.53, 0.13.54, 0.13.56, 0.13.57, 0.14.4, 0.14.12, 0.14.13, 0.14.15, 0.14.20, 0.14.31, 0.15.0, 0.15.3, 0.15.4, 0.15.5, 0.15.10, 0.16.17, 0.17.0, 0.17.1, 0.17.2, 0.17.4, 0.17.6, 0.17.7, 0.17.8, 0.18.1, 0.18.5, 0.18.9, 0.18.10, 0.18.14, 0.19.4, 0.19.8, 0.19.9, 0.19.10, 0.19.16
      ERROR: Could not find a version that satisfies the requirement polars>=0.26.0 (from versions: 0.8.10, 0.8.11b1, 0.8.11, 0.8.12, 0.8.13b1, 0.8.14, 0.8.15b1, 0.8.15, 0.8.16, 0.8.17, 0.8.18, 0.8.19, 0.8.20, 0.8.21b1, 0.8.21b2, 0.8.21, 0.8.22b1, 0.8.22, 0.8.23, 0.8.24, 0.8.25, 0.8.26, 0.8.27b1, 0.8.27b2, 0.8.27, 0.8.28, 0.8.29, 0.9.0b1, 0.9.0b2, 0.9.0, 0.9.1, 0.9.2, 0.9.3b1, 0.9.3, 0.9.4, 0.9.5b1, 0.9.5, 0.9.6b1, 0.9.6, 0.9.7, 0.9.8, 0.9.9, 0.9.10, 0.9.11, 0.9.12, 0.10.0, 0.10.1, 0.10.2, 0.10.3, 0.10.4, 0.10.5, 0.10.6, 0.10.7, 0.10.8, 0.10.9, 0.10.10, 0.10.11, 0.10.12, 0.10.13, 0.10.14, 0.10.15, 0.10.16, 0.10.17, 0.10.18, 0.10.19, 0.10.20, 0.10.21, 0.10.22, 0.10.23, 0.10.24, 0.10.25, 0.10.26, 0.10.27, 0.11.0, 0.11.1, 0.12.0, 0.12.1, 0.12.2, 0.12.3, 0.12.4, 0.12.5, 0.12.6, 0.12.7, 0.12.8, 0.12.9, 0.12.10, 0.12.11, 0.12.12, 0.12.13, 0.12.14, 0.12.15, 0.12.16, 0.12.17, 0.12.18, 0.12.19, 0.12.20, 0.12.21, 0.12.22, 0.12.23, 0.12.24, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.13.4, 0.13.5, 0.13.6, 0.13.7, 0.13.8, 0.13.10, 0.13.11, 0.13.12, 0.13.13, 0.13.14, 0.13.15, 0.13.16, 0.13.17, 0.13.18, 0.13.19, 0.13.21, 0.13.22, 0.13.23, 0.13.24, 0.13.25, 0.13.29, 0.13.31, 0.13.34, 0.13.36, 0.13.37, 0.13.38, 0.13.39, 0.13.40, 0.13.44, 0.13.46, 0.13.49, 0.13.50, 0.13.51, 0.13.52, 0.13.55, 0.13.58, 0.13.59, 0.13.61, 0.13.62, 0.14.0, 0.14.1, 0.14.2, 0.14.5, 0.14.6, 0.14.8, 0.14.9, 0.14.10, 0.14.11, 0.14.14, 0.14.17, 0.14.18, 0.14.19, 0.14.21, 0.14.23, 0.14.24, 0.14.25, 0.14.26, 0.14.27, 0.14.28, 0.14.29, 0.14.30, 0.15.1, 0.15.2, 0.15.6, 0.15.7, 0.15.8, 0.15.9, 0.15.11, 0.15.13, 0.15.14, 0.15.15, 0.15.16, 0.15.17, 0.15.18, 0.16.1, 0.16.2, 0.16.3, 0.16.4, 0.16.5, 0.16.6, 0.16.7, 0.16.8, 0.16.9, 0.16.10, 0.16.11, 0.16.12, 0.16.13, 0.16.14, 0.16.15, 0.16.16, 0.16.18, 0.17.3, 0.17.5, 0.17.9, 0.17.10, 0.17.11, 0.17.12, 0.17.13, 0.17.14, 0.17.15, 0.18.0, 0.18.2, 0.18.3, 0.18.4, 0.18.6, 0.18.7, 0.18.8, 0.18.11, 0.18.12, 0.18.13, 0.18.15, 0.19.0, 0.19.1, 0.19.2, 0.19.3, 0.19.5, 0.19.6, 0.19.7, 0.19.11, 0.19.12rc1, 0.19.12, 0.19.13rc1, 0.19.13, 0.19.14, 0.19.15, 0.19.17, 0.19.18, 0.19.19, 0.20.0, 0.20.1, 0.20.2, 0.20.3rc1, 0.20.3rc2, 0.20.3, 0.20.4, 0.20.5, 0.20.6rc1, 0.20.6, 0.20.7, 0.20.8)
      ERROR: No matching distribution found for polars>=0.26.0
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by MarcoGorelli** [comment]
Thanks @Chretien for the report, looks like a typo in the project.toml file. I'm surprised this didn't come up sooner
I'll fix and make a release first thing tomorrow

**Comment by MarcoGorelli** [comment]
hi @Chretien - I think it should be fixed now, could you try again please?

**Reply by Chretien** [reply]
Hey Marco, thanks for the quick response. I tried running the install again, and got a different error this time:

❯ pip install polars-xdt
WARNING: Ignoring invalid distribution -astapi (/Users/.../.pyenv/versions/3.9.10/lib/python3.9/site-packages)
Collecting polars-xdt
  Downloading polars_xdt-0.12.6.tar.gz (976 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 976.1/976.1 kB 13.9 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      Checking for Rust toolchain....
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Reply by MarcoGorelli** [reply]
thanks @Chretien for the report

I don't know, sorry, but I've asked on the Polars discord

Are you able to install Polars itself?

**Reply by Chretien** [reply]
Hey @MarcoGorelli , yes I am able to install (and very much enjoy) the standard Polars package. I currently have Polars version 0.20.7 installed.

**Reply by ritchie46** [reply]
The wheel for your specific version isn't available, so it tries to compile from source.

@MarcoGorelli I think you should check in polars CI if some wheels are missing.

**Reply by MarcoGorelli** [reply]
Thanks Ritchie! And thanks Chretien for your patience here, much appreciated

Right, I think I was just missing `"abi3-py38"` in the `pyo3` features. I've tried putting that, would you mind checking if it works now please @Chretien ? 🤞

**Reply by Chretien** [reply]
@MarcoGorelli @ritchie46 I have been able to successfully install polars-xdt-0.12.8. Thank you so much!!

**Reply by MarcoGorelli** [reply]
fantastic! thanks for helping improve the tool. I've updated the plugins cookiecutter to have the abi3 feature

closing then
